Sources/OutlineView/OutlineViewDataSource.swift

```diff
@@ -1,18 +1,67 @@
 import Cocoa
+import Combine
 
 @available(macOS 10.15, *)
-class OutlineViewDataSource<Data: Sequence>: NSObject, NSOutlineViewDataSource
-where Data.Element: Identifiable {
+class OutlineViewDataSource<Data: Sequence, Drop: DropReceiver>: NSObject, NSOutlineViewDataSource
+where Drop.DataElement == Data.Element {
     var items: [OutlineViewItem<Data>]
-
-    init(items: [OutlineViewItem<Data>]) {
+    var dropReceiver: Drop?
+    var dragWriter: DragSourceWriter<Data.Element>?
+    let childrenSource: ChildSource<Data>
+    
+    var treeMap: TreeMap<Data.Element.ID>
+    
+    private var dropExpansionNotificationToken: AnyCancellable? {
+        didSet {
+            // In case the original notification listener doesn't stop itself
+            // as set up in `acceptDrop...`, add this as a backup to end the
+            // notification listener after a short period.
+            if dropExpansionNotificationToken != nil {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+                    self.dropExpansionNotificationToken = nil
+                }
+            }
+        }
+    }
+    private var willExpandToken: AnyCancellable?
+    private var didCollapseToken: AnyCancellable?
+        
+    init(items: [OutlineViewItem<Data>], childSource: ChildSource<Data>) {
         self.items = items
+        self.childrenSource = childSource
+        
+        treeMap = TreeMap()
+        for item in items {
+            treeMap.addItem(item.value.id, isLeaf: item.children == nil, intoItem: nil, atIndex: nil)
+        }
+        
+        super.init()
+        
+        // Listen for expand/collapse notifications in order to keep TreeMap up to date
+        willExpandToken = NotificationCenter.default.publisher(for: NSOutlineView.itemWillExpandNotification)
+            .receive(on: DispatchQueue.main)
```

Comment:
This notification should be published on the main thread. https://github.com/Sameesunkaria/OutlineView/pull/10#discussion_r1089447551 
```suggestion
```